api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/BFFApiGatewayController.java

```diff
@@ -1065,6 +1065,14 @@ public Mono<Void> deleteInventoryByInventoryId(@PathVariable String inventoryId)
         return inventoryServiceClient.getLowStockProducts(inventoryId, stockThreshold);
     }
 
+    @SecuredEndpoint(allowedRoles = {Roles.ADMIN,Roles.INVENTORY_MANAGER,Roles.VET})
+    @GetMapping(value = "inventory/{inventoryId}/products/search")//, produces= MediaType.TEXT_EVENT_STREAM_VALUE)
+    public Flux<ProductResponseDTO> searchProducts(@PathVariable String inventoryId,
+                                                   @RequestParam(required = false) String productName,
+                                                   @RequestParam(required = false) String productDescription) {
+        return inventoryServiceClient.searchProducts(inventoryId, productName, productDescription);
+    }
+
```

Comment:
I would consider removing your endpoint from the BFFApiGatewayController since it's deprecated and we won't be using it anymore now that we have classes for each respective microservice. It doesn't hurt to have it in there for the moment but it will be something to think about.